File: rgb_to_hex.java
public static String rgbToHex(int r, int g, int b) {
    r = Math.min(255, Math.max(0, r));
    g = Math.min(255, Math.max(0, g));
    b = Math.min(255, Math.max(0, b));
    return String.format("%02X%02X%02X", r, g, b);
}

File: rgb_to_hex.cc
#include <iostream>
#include <sstream>
#include <iomanip>

std::string rgbToHex(int r, int g, int b)
{
    r = std::max(0, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    std::stringstream ss;
    ss << std::uppercase << std::hex << std::setfill('0')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    return ss.str();
}

//Test with std::string hexColor = rgbToHex(255, 127, 0); // returns "FF7F00"